File: C#/Basic logic/F2/F2_4.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace F2
{
    internal class F2_4
    {
        static void Mainx(string[] args)
        {
            int[] polecisla = new int[8];
            int cislo;
            int vystup, sub;
            string vypis = "";
            Random r = new Random();

            Console.WriteLine("zadej které číslo tě zajímá");
            cislo = int.Parse(Console.ReadLine());

            for (int i = 0; i <= polecisla.Length - 1; i++)
            {
                polecisla[i] = r.Next(1, 11);
                vypis = vypis + polecisla[i] + " ";
            }
            vystup = pocetvyskytuF(polecisla, cislo); //funkce
            pocetvyskytuS(polecisla, cislo, out sub); //subrutina

            Console.WriteLine("padlé čísla: {0}\nkolikrát padlo zadané číslo: {1}x", vypis, vystup); //funkce
            Console.WriteLine("padlé čísla: {0}\nkolikrát padlo zadané číslo: {1}x", vypis, sub); //subrutina
        }
        static int pocetvyskytuF(int[] policko, int pocet) //funkce
        {
            int x = 0;

            for (int i = 0; i <= policko.Length - 1; i++)
            {
                if (policko[i] == pocet)
                    x += 1;
            }
            return x;
        }

        static void pocetvyskytuS(int[] policko, int pocet, out int s) //subrutina
        {
            int x = 0;

            for (int i = 0; i <= policko.Length - 1; i++)
            {
                if (policko[i] == pocet)
                    x += 1;
            }
            s = x;
        }
    }
}

File: Visual Basic/VB - Logic exercises/F2/F2_4.vb
Module F2_4
    Sub main()
        Dim A(9) As Integer
        Dim C As Byte, i As Byte, kolikrat As Byte
        Dim r As New Random
        Dim vypis As String
        Dim soucet As Byte

        vypis = ""
        C = InputBox("zadej číslo 1-10")

        For i = 0 To 9
            A(i) = r.Next(1, 11)
            vypis = vypis + Str(A(i))
        Next

        kolikrat = pocetvyskytuF(A, C)
        pocetvyskytuS(A, C, soucet)
        MsgBox("losovaná čísla: " + vypis + Chr(10) + "číslo: " + Str(C) + " padlo (fce)" + Str(kolikrat) + "x" + Chr(10) + "číslo: " + Str(C) + " padlo (sbrtn)" + Str(soucet) + "x" + Chr(10))

    End Sub



    Function pocetvyskytuF(pole() As Integer, pocet As Byte) As Byte
        Dim i As Byte, soucet As Byte

        For i = 0 To 9
            If pole(i) = pocet Then
                soucet = soucet + 1
            End If
        Next
        pocetvyskytuF = soucet
    End Function



    Sub pocetvyskytuS(pole() As Integer, pocet As Byte, ByRef soucet As Byte)

        Dim i As Byte

        For i = 0 To 9
            If pole(i) = pocet Then
                soucet = soucet + 1
            End If
        Next

    End Sub

End Module
